smart_open/tests/test_smart_open.py

```diff
@@ -486,6 +485,19 @@ def test_s3_mode_mock(self, mock_write, mock_boto):
         mock_boto.connect_s3().get_bucket.assert_called_with("mybucket")
         self.assertTrue(mock_write.called)
 
+    @mock.patch('smart_open.smart_open_lib.subprocess')
+    def test_hdfs(self, mock_subprocess):
+        """Is HDFS write called correctly"""
+        smart_open_object = smart_open.HdfsOpenWrite(smart_open.ParseUri("hdfs:///tmp/test.txt"))
```

Comment:
Yes, agree that HdfsOpenRead doesn't have this test. There is benefit in adding a test like that though to replace the exception test. As an aside, an ideal would be to mock hdfs similarly to [s3 tests]( https://github.com/skibaa/smart_open/blob/0fe9e521b5f7cf3589f6c5b2c537acec875a765d/smart_open/tests/test_smart_open.py#L251)